services/ui-src/src/utils/autosave/autosave.ts

```diff
@@ -37,45 +35,34 @@ export const autosaveFieldData = async ({
   const { id, updateReport } = report;
   const { userName, state } = user;
 
-  // if field value hasn't changed from database value, don't autosave field
-  const onlyChangedFields = fields.filter(
-    // TODO: do we need a deeper equality check here?
-    (field: FieldInfo) => {
-      const { value, hydrationValue, defaultValue } = field;
-      return value !== defaultValue && !isEqual(value, hydrationValue);
-    }
+  // for each passed field, format for autosave payload (if changed)
+  const fieldsToSave: FieldDataTuple[] = await Promise.all(
+    fields
+      // filter to only fields with changed values
+      .filter((field: FieldInfo) => field.value !== field.hydrationValue)
+      // determine appropriate field value to set and return as tuple
+      .map(async (field: FieldInfo) => {
+        const { name, value, defaultValue, overrideCheck } = field;
+        const fieldValueIsValid = await form.trigger(name);
+        // if field value is valid or validity check overriden, use field value
+        if (fieldValueIsValid || overrideCheck) return [name, value];
```

Comment:
Note: The new `overrideCheck` prop can be used for:
- DynamicField: Where we want to ignore the result of a field validity check (to allow us to blank internal text inputs) and instead of falling back on a default value, just pass the actual value through.
- ChoiceListField nested children that we want to force-blank or reset when parent field is unchecked: Where we want to ignore the check and set it to a value we choose, which will always be the default value, but which we can just pass as the value (see the change to ChoiceListField).